proto/documents/v3alpha/documents.proto

```diff
@@ -110,6 +127,139 @@ message ListRootDocumentsResponse {
   string next_page_token = 2;
 }
 
+// Request to list accounts.
+message ListAccountsRequest {
+  // Optional. Number of results to return per page.
+  // If not set, the server will pick some default number.
+  int32 page_size = 1;
+
+  // Optional. Token of the page to request, obtained from
+  // a previous response.
+  string page_token = 2;
+
+  // Optional. If set to true, only accounts that the user is subscribed to will be returned.
+  // If at least one document in the account has a subscription, then the account is considered subscribed.
+  bool subscribed_only = 3;
+
+  // Sorting mode for the list response.
+  SortOrder sort_order = 4;
+
+  // Sorting options for the list of accounts.
+  enum SortOrder {
+    // Sorting by the most recent activity time.
+    LATEST_ACTIVITY_FIRST = 0;
+
+    // Sorting by the name of the account (i.e. title of the root document).
+    NAME_ALPHABETICAL = 1;
+  }
+}
+
+// Response to list accounts.
+message ListAccountsResponse {
+  // List of accounts.
+  repeated Account accounts = 1;
+
+  // Token for the next page if there're more results.
+  string next_page_token = 2;
+}
+
+// Aggregated information about an account.
+// Some of this information comes from the accounts' root document,
+// some is aggregated from the other documents inside of the account.
+message Account {
+  // ID of the account.
+  string id = 1;
+
+  // Metadata attributes from the root document of the account.
+  map<string, string> metadata = 2;
+
+  // Summary of the activity in the entire account.
+  ActivitySummary activity_summary = 3;
+
+  // Whether the user is subscribed to this account.
+  bool is_subscribed = 4;
+}
+
+// Request to list a directory.
+message ListDirectryRequest {
+  // Optional. Size of the page.
+  int32 page_size = 1;
+
+  // Optional. Pagination token.
+  string page_token = 2;
+
+  // Required. ID of the account to list the directory for.
+  string account = 3;
+
+  // Required. Path of the directory to list documents for.
+  // Empty string means root directory.
+  // The root document is considered a part of the root directory.
```

Comment:
>The root document is considered a part of the root directory.

I'm not sure about this, we always throw away the root doc when it appears in our lists

Reply:
Hm... OK. Won't put it there then.

Reply:
I removed this comment. Although this sounds a bit contradictory to what you said here, I guess? https://github.com/seed-hypermedia/seed/pull/71#discussion_r1871751811